File: factorial.cpp
#include <bits/stdc++.h>
using namespace std;

int factorial(int n){

    //base case
    if(n == 0)
    return 1;

    return n * factorial(n - 1);
}

int main() {

    int n;
    cin >> n;

    int ans = factorial(n);
    cout << ans << endl;

    return 0;
}

File: JS/Factorial.js
function factorial(n) {
    //base case for 0!
    if (n == 0 || n == 1) {
        return 1;
        //recursion case
    } else {
        return n * factorial(n - 1);
    }
}
//factorial of 5
let n = 5;
answer = factorial(n)

console.log(answer);
